### PR TITLE
Create ADNLPModel and change SimplNLPModel to be user defined only

### PR DESCRIPTION
**Pull request by abelsiqueira**
A very large PR, so a little explaining helps:

- ADNLPModel is the old SimpleNLPModel. It should be only a name change.
- SimpleNLPModel receives each NLPModels function (minus the nlp parameter).
- Then fixing every test to make it work.
- Finally the docs were updated, including change to orange.

### CODE REVIEW

**Comment by abelsiqueira** [comment]
Related to #36

File: docs/src/index.md [hunk 1]
```diff
@@ -40,13 +42,31 @@ Such instances are composed of
   including the number of variables, constraints, bounds on the variables, etc.
 - other data specific to the provenance of the problem.
 
+## Install
+
+The current usable version of NLPModels.jl is the development one.
```
Comment (dpo) [inline docs/src/index.md]: "is in the development branch"

File: docs/src/index.md [hunk 2]
```diff
@@ -40,13 +42,31 @@ Such instances are composed of
...
- - [`SimpleNLPModel`](@ref): Uses
+ - [`ADNLPModel`](@ref): Uses
    [`ForwardDiff`](http://github.com/JuliaDiff/ForwardDiff.jl) to compute the
-   derivatives. It has a very simple interface.
+   derivatives. It has a very simple interface, though it isn't very efficient
+   for larger problems.
  - [`JuMPNLPModel`](@ref): Uses a [`JuMP`](https://github.com/JuliaOpt/JuMP.jl) model.
-  - [`SlackModel`](@ref): Creates an equality constrained problem with bounds
+ - [`SimpleNLPModel`](@ref): Only uses user defined functions. For complete
+   control of the user.
```
Comment (dpo) [inline docs/src/index.md]: I think you can remove the second sentence here ("For complete...")

File: src/autodiff_model.jl [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+end
+
+function cons(nlp :: ADNLPModel, x :: Vector)
+  nlp.counters.neval_cons += 1
+  return nlp.c(x)
+end
+
+function cons!(nlp :: ADNLPModel, x :: Vector, c :: Vector)
+  nlp.counters.neval_cons += 1
+  c[:] = nlp.c(x)
```
Comment (dpo) [inline src/autodiff_model.jl]: Should this be `c[1:nlp.meta.nvar]` to avoid situations like we had recently?

**Comment by abelsiqueira** [comment]
Fixed. Nightly OSX was broken due to curl timeout, so I restarted.

**Comment by dpo** [comment]
Ready with this?

**Comment by abelsiqueira** [comment]
Yes

**Pull request merged**